src/mono/System.Private.CoreLib/src/System/Reflection/FieldInfo.Mono.cs

```diff
@@ -24,7 +24,7 @@ public static FieldInfo GetFieldFromHandle(RuntimeFieldHandle handle, RuntimeTyp
                 throw new ArgumentException(SR.Argument_InvalidHandle);
             FieldInfo fi = internal_from_handle_type(handle.Value, declaringType.Value);
             if (fi == null)
-                throw new ArgumentException("The field handle and the type handle are incompatible.");
+                throw new ArgumentException(SR.Argument_FieldPropertyEventAndTypeHandleIncompatibility);
```

Comment:
Can this be `Argument_ResolveFieldHandle` like in CoreCLR?

Reply:
Theoretically yes but I don't have the type(s) to fill in the format string and `RuntimeFieldHandle.GetApproxDeclaringType` is only implemented in CoreCLR.

I'd rather keep this as is for now so we can finally get this PR in.